block/manager.go

```diff
@@ -1500,27 +1506,24 @@ func (m *Manager) execCreateBlock(_ context.Context, height uint64, lastSignatur
 			BaseHeader: types.BaseHeader{
 				ChainID: lastState.ChainID,
 				Height:  height,
-				Time:    uint64(timestamp.UnixNano()), //nolint:gosec
+				Time:    uint64(batchData.Time.UnixNano()), //nolint:gosec // why is time unix? (tac0turtle)
 			},
-			DataHash:        make(types.Hash, 32),
+			DataHash:        batchdata,
 			ConsensusHash:   make(types.Hash, 32),
 			AppHash:         lastState.AppHash,
 			ProposerAddress: m.genesis.ProposerAddress,
 		},
 		Signature: *lastSignature,
 	}
 
-	data := &types.Data{
-		Txs: make(types.Txs, len(txs)),
-		// IntermediateStateRoots: types.IntermediateStateRoots{RawRootsList: nil},
-		// Note: Temporarily remove Evidence #896
-		// Evidence:               types.EvidenceData{Evidence: nil},
+	blockData := &types.Data{
+		Txs: make(types.Txs, 0, len(batchData.Batch.Transactions)),
```

Comment:
```suggestion
		Txs: make(types.Txs, len(batchData.Batch.Transactions)),
```
You're accessing by index in the loop below, need to allocate size not capacity, otherwise it panics.